tests/test_types.py

```diff
@@ -134,3 +135,147 @@ def test_serialize_arc():
     assert abs(new_arc.start_angle - arc.start_angle) < 1e-6
     assert abs(new_arc.end_angle - arc.end_angle) < 1e-6
     assert new_arc.circle.radius == arc.circle.radius
+
+
+def test_serialize_sphere():
+    from compas.geometry import Sphere
+
+    sphere = Sphere(radius=2.0, frame=Frame.worldXY())
+
+    bts = pb_dump_bts(sphere)
+    new_sphere = pb_load_bts(bts)
+
+    assert isinstance(new_sphere, Sphere)
+    assert new_sphere.radius == sphere.radius
+    assert new_sphere.frame.point == sphere.frame.point
+
+
+def test_serialize_cylinder():
+    from compas.geometry import Cylinder
+
+    cylinder = Cylinder(radius=1.5, height=3.0, frame=Frame.worldXY())
+
+    bts = pb_dump_bts(cylinder)
+    new_cylinder = pb_load_bts(bts)
+
+    assert isinstance(new_cylinder, Cylinder)
+    assert new_cylinder.radius == cylinder.radius
+    assert new_cylinder.height == cylinder.height
+    assert new_cylinder.frame.point == cylinder.frame.point
+
+
+def test_serialize_cone():
+    from compas.geometry import Cone
+
+    cone = Cone(radius=1.0, height=2.5, frame=Frame.worldXY())
+
+    bts = pb_dump_bts(cone)
+    new_cone = pb_load_bts(bts)
+
+    assert isinstance(new_cone, Cone)
+    assert new_cone.radius == cone.radius
+    assert new_cone.height == cone.height
+    assert new_cone.frame.point == cone.frame.point
+
+
+def test_serialize_torus():
+    from compas.geometry import Torus
+
+    torus = Torus(radius_axis=2.0, radius_pipe=0.5, frame=Frame.worldXY())
+
+    bts = pb_dump_bts(torus)
+    new_torus = pb_load_bts(bts)
+
+    assert isinstance(new_torus, Torus)
+    assert new_torus.radius_axis == torus.radius_axis
```

Comment:
use the comparison methods of compas.tolerance.TOL (Tolerance) to compare floats